test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/ODataUriParserTests.cs

```diff
@@ -1035,5 +1037,95 @@ public void ParseFullyQualifiedEnumWithResolver()
             Uri resultUri = uri.BuildUri(ODataUrlKeyDelimiter.Parentheses);
             Assert.Equal(fullUriString, Uri.UnescapeDataString(resultUri.OriginalString));
         }
+
+        [Fact]
+        public void ParseEscapeFunctionUrlReturnsTheSameODataPath()
+        {
+            // Arrange
+            IEdmModel model = GetEdmModelWithEscapeFunction(escape: true);
+
+            // Act
+            string fullUriString = ServiceRoot + "/root/NS.Function(path='photos%2F2018%2FFebruray')";

```

Comment:
>"/root/NS.Function(path='photos%2F2018%2FFebruray') [](start = 49, length = 51)

Can we add tests for the escape function followed by a nav prop, key segment, additional escape path, or regular function?

Reply:
Thanks for all of the added tests!

---
In reply to: [255706009](https://github.com/OData/odata.net/pull/1400#discussion_r255706009) [](ancestors = 255706009)